### PR TITLE
Cleanup string formatting.

### PR DESCRIPTION
**Pull request by knotapun**
This commit attempts to clean out old cruft, to make room for new cruft. This is clearly not the most pressing issue in the world, however I think it's far more legible. 

This change consists of:
- Replacing most %-strings with f-strings.
- Cleanup a few areas where repeated string formatting logic was used, by using a loop instead.

### CODE REVIEW

**Comment by knotapun** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by knotapun** [comment]
recheck

**Comment by knotapun** [comment]
Read through the action result, and it looks like it wasn't happy about an extra line, and I forgot to wrap arguments for the SQLite call in the sample, so it did a different form of injection. Ironic. At least it failed fast?

File: docs/recipes-images.rst [hunk 1]
```diff
@@ -525,8 +525,8 @@ This script creates an approximate image of it as a PNG, by going down to one-pi
     t1 = time.perf_counter()
     pm.save("sierpinski-punch.png")
     t2 = time.perf_counter()
-    print ("%g sec to create / fill the pixmap" % round(t1-t0,3))
-    print ("%g sec to save the image" % round(t2-t1,3))
+    print ("{round(t1-t0,3)} sec to create / fill the pixmap")
+    print ("{round(t2-t1,3)} sec to save the image")
```
[the remark this answers is not in the document]
Comment (knotapun) [reply docs/recipes-images.rst]: Valid take. Changed.

**Comment by knotapun** [comment]
> * The `raise` syntax is invalid here. It should be `raise ValueError(f"max. code point found: {ord(max(text))}, increase limit")` without the colon after `ValueError`.
> 
> ```
>                raise ValueError:(f"max. code point found: {ord(max(text))}, increase limit")
> ```
> 
> **src/utils.py:1640**

The syntax error is real, but I didn't introduce it. That's easy to fix though.

**Comment by knotapun** [comment]
Everything copilot complained about, (Except the pass thing, which it was wrong about), I fixed. I don't think the dictionary thing makes a difference speed wise, but I think it's stylistically better.

**Comment by jamie-lemon** [comment]
@knotapun Would it be possible to squash your commits here?

**Comment by knotapun** [comment]
They're now squashed into one commit. Git is my biggest weakness, sorry if this is not the appropriate way to do this.

**Comment by jamie-lemon** [comment]
Looks good to me, my "git-fu" is also pretty weak when it comes to this sort of thing!

As far as a I am concerned this PR is fine, but I think we need to spend some time double checking and testing that none of our tests are broken (which I doubt). So don't be surprised if it takes a few days :)

**Comment by julian-smith-artifex-com** [comment]
@knotapun In #4900, i've separately merged the changes in this PR into the current tree, except for the changes to `src/utils.py`. I've preserved your authorship of the commit.

The changes to `src/utils.py` conflicted i think because of code being moved to `src/__init__.py`. I added a separate commit that fixes the remaining use of %-strings in `src/__init__.py`.

Many thanks for creating this PR, it's much appreciated!
